### ISSUE TITLE
03-basics: Error creating the database instance 

### ISSUE THREAD

**Issue by carlos-salinas** [sidpalas/devops-directive-terraform-course#8]
Executing ```terraform apply``` in the **web-app** folder the database instance provisioning fails: 
```
aws_db_instance.db_instance: Creating...
╷
│ Error: Error creating DB Instance: InvalidParameterCombination: Cannot find version 9.5 for postgres
│       status code: 400, request id: 9085a882-f29c-4d65-a328-597a9ffa17ba
│ 
│   with aws_db_instance.db_instance,
│   on main.tf line 200, in resource "aws_db_instance" "db_instance":
│  200: resource "aws_db_instance" "db_instance" {
│ 
```

**Comment by sidpalas** [comment]
Do you have a requirement to use 9.5? I don't think RDS supports it anymore.

https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_PostgreSQL.html#PostgreSQL.Concepts.General.DBVersions

**Comment by carlos-salinas** [comment]
Correct! I modified manually the previous snippet trying to put back ```12.5``` and incorrectly I set ```9.5``` because I tried with several versions and none of then worked. Moreover, I checked the version drop-down at AWS Console and seems ```12.5``` is not available anymore.

**Comment by DmitriKonnovNN** [comment]
Unfortunately, available DB-versions varies in RDS-service from region to region. This confuses.

**Comment by MarcPadrosJimenez** [comment]
In relation to this open issue, I am dealing with the following error: 
` Error: Error creating DB Instance: InvalidParameterCombination: RDS does not support creating a DB instance with the following combination: DBInstanceClass=db.t2.micro, Engine=postgres, EngineVersion=12.19, LicenseModel=postgresql-license. For supported combinations of instance class and database engine version, see the documentation.
│       status code: 400, request id: dd8cce7e-1c08-4a4b-9a73-580b58af8955
│
│   with aws_db_instance.db_instance,
│   on main.tf line 205, in resource "aws_db_instance" "db_instance":
│  205: resource "aws_db_instance" "db_instance" {`

How can I solve it? I checked for AWS documentation, but I didn't find which is a compatible PostgreSQL version for db.t2.micro.
I hope someone can give me any clue.
Thank you.

**Comment by SaadHadadia** [comment]
I have encountered also encountered the same error as  @MarcPadrosJimenez

> In relation to this open issue, I am dealing with the following error: ` Error: Error creating DB Instance: InvalidParameterCombination: RDS does not support creating a DB instance with the following combination: DBInstanceClass=db.t2.micro, Engine=postgres, EngineVersion=12.19, LicenseModel=postgresql-license. For supported combinations of instance class and database engine version, see the documentation. │ status code: 400, request id: dd8cce7e-1c08-4a4b-9a73-580b58af8955 │ │ with aws_db_instance.db_instance, │ on main.tf line 205, in resource "aws_db_instance" "db_instance": │ 205: resource "aws_db_instance" "db_instance" {`
> 
> How can I solve it? I checked for AWS documentation, but I didn't find which is a compatible PostgreSQL version for db.t2.micro. I hope someone can give me any clue. Thank you.

AWS has begun phasing out `t2.micro` for PostgreSQL engines, meaning newer versions like 12.x aren’t supported on that instance class.

Switching to `db.t3.micro` fixed it because that instance type supports PostgreSQL 12.

### Sources

- [RDS InvalidParameterCombination - Combination working before](https://repost.aws/questions/QUdYK-DiggRBOsNFSsvqVacA/rds-invalidparametercombination-combination-working-before)
- [Adding DB to my RDS I do not see t2.micro or t3.micro options](https://repost.aws/questions/QUx2Wv6CxJSH26t1K-DaC6fg/adding-db-to-my-rds-i-do-not-see-t2-micro-or-t3-micro-options)
